### DISCUSSION TITLE
Android-release build (proguard rules)

### DISCUSSION THREAD

**Discussion by antailyaqwer** [ionspin/kotlin-multiplatform-libsodium#39] [Show and tell]
Maybe this must be somewhere in the readme file

To run android build with minification enabled we shoud add some rules to `proguard-rules.pro` file:

```
-keep class com.ionspin.kotlin.crypto.** { *; }
-keep class com.sun.jna.** { *; }
```

**Comment by edenman** [comment]
I'm getting this when I try these proguard rules.  Did you run into anything like this?
```
ERROR: Missing classes detected while running R8. Please add the missing classes or apply additional keep rules that are generated in /Users/edenman/workspace/x-android/app/build/outputs/mapping/release/missing_rules.txt.
ERROR: R8: Missing class java.awt.Component (referenced from: long com.sun.jna.Native$AWT.getComponentID(java.lang.Object) and 3 other contexts)
Missing class java.awt.GraphicsEnvironment (referenced from: long com.sun.jna.Native$AWT.getComponentID(java.lang.Object))
Missing class java.awt.HeadlessException (referenced from: long com.sun.jna.Native$AWT.getComponentID(java.lang.Object))
Missing class java.awt.Window (referenced from: long com.sun.jna.Native$AWT.getWindowID(java.awt.Window) and 2 other contexts)
```

**Reply by antailyaqwer** [reply]
I haven't been using this lib for a long time.
Try add keep rule for java.awt:
`-keep class java.awt.** { *; }`

**Reply by edenman** [reply]
Thanks for the quick response! No luck, same error.  I wouldn't expect there to be any AWT stuff in our app though, so it makes sense that it's missing.

**Reply by ionspin** [reply]
Are you using multiplatform compose?

**Reply by edenman** [reply]
nope, just KMP

**Reply by ionspin** [reply]
Ah, I didn't see it was from JNA. I guess you could add dontWarn for those classes `-dontwarn java.awt.HeadlessException`

**Reply by ionspin** [reply]
Here's the source of JNA Native.java with those classes if you are interested in why they pop up https://github.com/java-native-access/jna/blob/master/src/com/sun/jna/Native.java

**Reply by edenman** [reply]
Nice.  That works!  full set of proguard rules:
```
-keep class com.ionspin.kotlin.crypto.** { *; }
-keep class com.sun.jna.** { *; }
-dontwarn java.awt.**
```
